src/data.ts

```diff
@@ -240,30 +267,31 @@ export class Data {
 				}) as IProfile);
 			}
 		});
-		profiles.forEach((it) => {
-			db.profiles.put(it);
-		});
-		// db.profiles.bulkPut(profiles)
+
+		queueMicrotask(()=>{
+			db.profiles.bulkPut(profiles)
+		})
+		await yieldMicrotask()
 	}
 
 	private rereceivedCounter = 0
-	private async onEvent(event: IEvent, relay: string): void {
-		if (!Data.instance.allEventIds.has(event.id)) {
-			Data.instance.events.push(event);
-			Data.instance.allEventIds.add(event.id);
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	private async onEvent(event: IEvent, relay: string): Promise<void> {
+		if (!Data.allEventIds.has(event)) {
+			Data.allEventIds.add(event);
```

Comment:
allEventIds was meant to keep track of events by their IDs. Thus the name. I'm not sure what you made of this now but if the logic changed, the name probably should also change.